src/bindings/python/flux/job.py

```diff
@@ -1097,3 +1102,100 @@ def from_command(
         tasks = [{"command": command, "slot": "task", "count": task_count_dict}]
         attributes = {"system": {"duration": 0}}
         return cls(resources, tasks, attributes=attributes)
+
+    @classmethod
+    def from_batch_command(
+        cls,
+        script,
+        jobname,
+        args=None,
+        num_slots=1,
+        cores_per_slot=1,
+        gpus_per_slot=None,
+        num_nodes=None,
+        broker_opts=None,
+    ):
+        """Create a Jobspec describing a nested Flux instance controlled by a script.
+
+        The nested Flux instance will execute the script with the given
+        command-line arguments after copying it and setting the executable bit.
+        Conceptually, this differs from the `from_nest_command`, which also creates a
+        nested Flux instance, in that it a) requires the initial program of the new
+        instance to be an executable text file and b) creates the initial program
+        from a string rather than using an executable existing somewhere on the
+        filesystem.
+
+        Use setters to assign additional properties.
+
+        :param script: contents of the script to execute, as a string. The
+            script should have a shebang (e.g. `#!/bin/sh`) at the top.
+        :param jobname: the name of the job, for Flux to record internally
```

Comment:
This is minor, but this parameter might be confused with the optional job name set in `system.job.name` of the jobspec.

Maybe just reword the description to something like `The name to use as the argv[0] for this job. This will be the default job name reported by Flux. (The actual argv is overridden by the job shell when executed)`